huaweicloud/services/live/resource_huaweicloud_live_transcoding.go

```diff
@@ -0,0 +1,306 @@
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceTranscoding() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceTranscodingCreate,
+		UpdateContext: resourceTranscodingUpdate,
+		DeleteContext: resourceTranscodingDelete,
+		ReadContext:   resourceTranscodingRead,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"domain_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"app_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"video_encoding": { //codec
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"H264", "H265"}, false)),
+			},
+
+			"templates": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 4,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringMatch(regexp.MustCompile("[a-zA-Z0-9-]{1,64}$"),
+								"The name can contain a maximum of 64 characters, and only contains letters,"+
+									" digits and hyphens (-)."),
+						},
+
+						"width": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+
+						"height": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+
+						"bitrate": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(40, 30000),
+						},
+
+						"frame_rate": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntBetween(0, 30),
+							Computed:     true,
+						},
+					},
+				},
+			},
+
+			"low_bitrate_hd": { //hdlb
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"on", "off"}, false),
+				Computed:     true,
+			},
+		},
+	}
+}
+
+func resourceTranscodingCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcLiveV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Live v1 client: %s", err)
+	}
+
+	transcodingParams, err := buildTranscodingParams(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	createOpts := &model.CreateTranscodingsTemplateRequest{
+		Body: transcodingParams,
+	}
+	log.Printf("[DEBUG] Create Live transcoding params : %#v", createOpts)
```

Comment:
suggest to delete the space between params and :

Reply:
fix